core/src/main/kotlin/com/strumenta/kolasu/transformation/Transformation.kt

```diff
@@ -366,7 +353,7 @@ open class ASTTransformer(
         val childFactory = childNodeFactory as ChildNodeFactory<Any, Any, Any>
         val childrenSource = childFactory.get(getSource(node, source))
         val child: Any? = if (pd.multiple) {
-            (childrenSource as List<*>).map { transformIntoNodes(it, node) }.flatten()
+            (childrenSource as List<*>?)?.map { transformIntoNodes(it, node) }?.flatten()
```

Comment:
I think lists of children should never be null. Do we have a reason for allowing null lists?

Reply:
Sorry, this wasn't meant for this PR. However, yes there is a use case. I encountered it in the SAS parser. If I remember correctly, in case of some parse errors, ANTLR can return null collections, and the transformer would crash.
However, probably it's better that the transformer always returns a non-null list.